src/Fixer/Basic/NonPrintableCharacterFixer.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Basic;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Removes Zero-width space (ZWSP), Non-breaking space (NBSP) and other invisible unicode symbols.
+ *
+ * @author Ivan Boprzenkov <ivan.borzenkov@gmail.com>
+ */
+final class NonPrintableCharacterFixer extends AbstractFixer
+{
+    private $symbolsReplace;
+
+    private static $tokens = array(
+        T_STRING,
+        T_STRING_VARNAME,
+        T_INLINE_HTML,
+        T_WHITESPACE,
+        T_COMMENT,
+        T_ENCAPSED_AND_WHITESPACE,
+        T_CONSTANT_ENCAPSED_STRING,
+        T_END_HEREDOC,
+        T_OPEN_TAG,
+        T_DOC_COMMENT,
```

Comment:
I see you added test for `T_DOC_COMMENT`. that's nice, but following are still missing:
```
        T_STRING,
        T_STRING_VARNAME,
        T_INLINE_HTML,
        T_WHITESPACE,
        T_ENCAPSED_AND_WHITESPACE,
        T_END_HEREDOC,
        T_OPEN_TAG,
        T_START_HEREDOC,
```